### PR TITLE
[Connector API] Implement update last_sync endpoint

### PR DESCRIPTION
**Pull request by jedrazb**
## Changes

- Implement `PUT _connector/{connector_id}/_last_sync` endpoint
- This endpoint expects body with a stats of a last sync, it overwrites the existing fields related to the last sync in a given connector doc

## Verification
- Add unit tests for index service
- Add BWC tests
- Add YML tests

Also, verified new endpoint manually.

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorLastSyncStatsAction.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public class RestUpdateConnectorLastSyncStatsAction extends BaseRestHandler {
+
+    @Override
+    public String getName() {
+        return "connector_update_last_sync_stats_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(PUT, "/" + EnterpriseSearch.CONNECTOR_API_ENDPOINT + "/{connector_id}/_last_sync"));
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorLastSyncStatsAction.java:29]: I started to extract `connector_sync_job_id` to a public constant `ParseField`, so it's easier to change it at multiple places, if we need to later. We could also do that with `connector_id` to remove some repetitions, but ofc nothing blocking 🙏

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorLastSyncStatsAction.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        public ConnectorSyncInfo getSyncInfo() {
+            return syncInfo;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorId)) {
+                validationException = addValidationError("[connector_id] cannot be null or empty.", validationException);
```
Comment (timgrein) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorLastSyncStatsAction.java:78]: Same here, this _could_ live inside a shared class, but also not blocking. We could also create a refactoring issue for these things, if we want to.

**Review by timgrein — APPROVED** [review]
LGTM! 🚀

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorLastSyncStatsAction.java (hunk 1 above)
Comment (jedrazb) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/RestUpdateConnectorLastSyncStatsAction.java:29]: Yeah good, point, I will add that (and your comment below) into a followup task to clean up the codebase a bit

**Pull request merged**